components/ai_chat/core/browser/engine/oai_api_client.cc

```diff
@@ -122,6 +125,22 @@ void OAIAPIClient::PerformRequest(
   }
 }
 
+std::optional<std::string> OAIAPIClient::ExtractErrorMessage(
+    const std::string& result) {
+  auto value = base::JSONReader::Read(result);
+
+  if (!value.has_value() && !value->is_dict()) {
+    return std::nullopt;
+  }
+
+  auto* error = value->GetDict().FindDict("error");
```

Comment:
Looks like there's a `type` property on the error message too - do we care about that for more detail?

Reply:
not as of now, nope. we just want surface error messages to the UI.